File: eventTester.py
from bs4 import BeautifulSoup
import json
import re

with open("events.html", "r", encoding="utf-8") as f:
    soup = BeautifulSoup(f, "html.parser")

event_data = []

for a_tag in soup.select('a[href*="/meets/"]'):
    href = a_tag.get("href", "")
    match = re.match(r"/meets/\d+/events/(relay|individual)/(\d+)", href)
    if not match:
        continue

    category, event_id = match.groups()
    name_tag = a_tag.select_one("h6")
    grade_tag = a_tag.select_one("app-event-name-secondary div")
    name = name_tag.get_text(strip=True) if name_tag else "Unknown Event"
    grade = grade_tag.get_text(strip=True) if grade_tag else ""

    full_name = f"{name} {grade}".strip()
    event_data.append({
        "id": event_id,
        "category": category,
        "name": full_name
    })

# Save it to a JSON file
with open("event_list.json", "w", encoding="utf-8") as f:
    json.dump(event_data, f, indent=2)

print(f"✅ Extracted {len(event_data)} events and saved to event_list.json")

File: eventTester.js
const fs = require('fs');
const cheerio = require('cheerio');

const html = fs.readFileSync('events.html', 'utf-8');
const $ = cheerio.load(html);

const eventData = [];

$('a[href*="/meets/"]').each((_, aTag) => {
    const href = $(aTag).attr('href') || '';
    const match = href.match(/\/meets\/\d+\/events\/(relay|individual)\/(\d+)/);
    if (!match) return;

    const [, category, eventId] = match;
    const nameTag = $(aTag).find('h6').first();
    const gradeTag = $(aTag).find('app-event-name-secondary div').first();
    const name = nameTag.text().trim() || "Unknown Event";
    const grade = gradeTag.text().trim() || "";

    const fullName = `${name} ${grade}`.trim();
    eventData.push({
        id: eventId,
        category: category,
        name: fullName
    });
});

fs.writeFileSync('event_list.json', JSON.stringify(eventData, null, 2), 'utf-8');
console.log(`✅ Extracted ${eventData.length} events and saved to event_list.json`);
